core-restore/src/main/scala/io/aiven/guardian/kafka/restore/RestoreClientInterface.scala

```diff
@@ -86,36 +88,36 @@ trait RestoreClientInterface[T <: KafkaProducerInterface] extends LazyLogging {
       case None => true
     }
 
-  private[kafka] def restoreKey(key: String): Future[Done] = {
+  private[kafka] def restoreKey(key: String): Source[ByteString, NotUsed] = {
     val source = Source
       .single(key)
       .via(downloadFlow)
 
-    val sourceWithCompression = BackupObjectMetadata.fromKey(key).compression match {
+    BackupObjectMetadata.fromKey(key).compression match {
       case Some(Gzip) => source.via(Compression.gunzip())
       case None       => source
     }
+  }
 
-    val base = sourceWithCompression
-      .via(CirceStreamSupport.decode[Option[ReducedConsumerRecord]](AsyncParser.UnwrapArray))
+  def restore: RunnableGraph[(UniqueKillSwitch, Future[Done])] = {
+    val sourceWithCompression = Source.future(finalKeys).flatMapConcat { keys =>
+      keys.map(key => restoreKey(key)) match {
```

Comment:
For some reason Akka does not have a `Source.combine` that just takes a list, instead `Source.combine` is an overloaded method with only 1 variation accepting a `List[Source]` as a vararg as its last parameter (with the first 2 parameters being a single source). See https://doc.akka.io/docs/akka/current/stream/operators/Source/combine.html